vdr/store/store.go

```diff
@@ -126,16 +126,18 @@ func (s *store) Update(id did.DID, current hash.SHA256Hash, next did.Document, m
 		var version int
 		var prevMetaRef []byte
 		latestRef, err := latestWriter.Get(stoabs.BytesKey(didString))
-		if err != nil {
-			return err
-		}
-
 		// check for existence
-		if latestRef == nil {
+		if errors.Is(err, stoabs.ErrKeyNotFound) {
 			return vdr.ErrNotFound
 		}
+		if err != nil {
+			return err
+		}
 
 		latestBytes, err := metadataWriter.Get(stoabs.BytesKey(latestRef))
+		if errors.Is(err, stoabs.ErrKeyNotFound) {
+			return vdr.ErrNotFound // TODO: is this correct? existence of a latestRef ~10 lines up means that this error cannot occur
```

Comment:
Is from a different shelf, so it would indicate corrupt storage

Reply:
so it should probably not return vdr.ErrNotFound since that indicates the DID does not exist and is handled different than the original error?

Reply:
Yes, probably just return a specific error?

Reply:
now returns 
```
fmt.Errorf("latest document metadata missing for %s (metadataShelfRef=%s)", didString, latestRef)
```